DQMServices/FwkIO/plugins/DQMRootSource.cc

```diff
@@ -42,204 +35,178 @@
-    return iStore.book1S(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TH1S* iHist) { mergeTogether(iElement->getTH1S(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TH1D* iHist) {
-    return iStore.book1DD(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TH1D* iHist) { mergeTogether(iElement->getTH1D(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TH2F* iHist) {
-    return iStore.book2D(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TH2F* iHist) { mergeTogether(iElement->getTH2F(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TH2S* iHist) {
-    return iStore.book2S(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TH2S* iHist) { mergeTogether(iElement->getTH2S(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TH2D* iHist) {
-    return iStore.book2DD(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TH2D* iHist) { mergeTogether(iElement->getTH2D(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TH3F* iHist) {
-    return iStore.book3D(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TH3F* iHist) { mergeTogether(iElement->getTH3F(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TProfile* iHist) {
-    return iStore.bookProfile(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TProfile* iHist) { mergeTogether(iElement->getTProfile(), iHist); }
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, TProfile2D* iHist) {
-    return iStore.bookProfile2D(iName, iHist);
-  }
-  void mergeWithElement(MonitorElement* iElement, TProfile2D* iHist) {
-    mergeTogether(iElement->getTProfile2D(), iHist);
-  }
-
-  MonitorElement* createElement(DQMStore& iStore, const char* iName, Long64_t& iValue) {
-    MonitorElement* e = iStore.bookInt(iName);
-    e->Fill(iValue);
-    return e;
-  }
+      // Check now labels sizes  are the same
+      if (l1->GetSize() != l2->GetSize()) {
+        return false;
+      }
 
-  //NOTE: the merge logic comes from DataFormats/Histograms/interface/MEtoEDMFormat.h
-  void mergeWithElement(MonitorElement* iElement, Long64_t& iValue) {
-    const std::string& name = iElement->getFullname();
-    if (name.find("EventInfo/processedEvents") != std::string::npos) {
-      iElement->Fill(iValue + iElement->getIntValue());
-    } else if (name.find("EventInfo/iEvent") != std::string::npos ||
-               name.find("EventInfo/iLumiSection") != std::string::npos) {
-      if (iValue > iElement->getIntValue()) {
-        iElement->Fill(iValue);
+      for (int i = 1; i <= a1->GetNbins(); ++i) {
+        TString label1 = a1->GetBinLabel(i);
+        TString label2 = a2->GetBinLabel(i);
+        if (label1 != label2) {
```

Comment:
String copies seem unnecessary.

Reply:
This code is taken unchanged from the old implementation, and I would not take any bets around root objects. 

`GetBinLabel` returns `const char*`, so making proper string object sounds like a slow but reasonable way to avoid `strcmp` and friends...

Reply:
Creating string objects just to make a comparison is never "reasonable", and using `TString` outside of ROOT code is almost never reasonable in itself.
 
Use `strcmp` directly, or use `std::string_view" for a higher level interface.

Reply:
Well, Fede added that code back in 2014 (#3173)... but yes, I'll try `string_view` once I get to it.